### ISSUE TITLE
.Net Standard 2.0 version

### ISSUE THREAD

**Issue by melucas** [rdavisau/sockets-for-pcl#149]
Any thoughts on converting this project to .Net Standard 2.0 or .Net Core now that MS has depreciated the portable projects?

**Comment by rdavisau** [comment]
Yes, I do have plans to switch it over in the near future. That said, I had expected it would be possible to use sockets directly from netstandard2, though I hadn't verified that.

**Comment by melucas** [comment]
So far it a is working as expect.  However, the complier moans about back converting the assembly to .Net 4.x.  I have tried it in a xamarin forms app running as Universal Windows and on an iPhone and my UDP connection are working and returning data.

**Comment by alanu86** [comment]
are there any new information about porting to netstandard2.0 ?

**Comment by newdeal30** [comment]
That feature would be absolutly great!

**Comment by dog44wgm** [comment]
It was all working beautifully for me until I tried to reference the Socket property in Release mode.

    // Should not be called on a TcpSocketClient that is not already connected.
    if (m_oClient.Socket.Connected)
        await m_oClient.DisconnectAsync();

when I got this

> warning : ILTransform : warning ILT0003: Method 'Communicator.<DisconnectAsync>d__10.MoveNext()' will always throw an exception due to the missing method 'TcpSocketClient.get_Socket()' in assembly 'Sockets.Plugin'.

so it can build and deploy fine, but then yes, it throws.
The problem doesn't exist in Debug.

**Comment by dog44wgm** [comment]
Sorry, I forgot to mention that the problem exists in UWP only.
